src/NuGet.Clients/NuGet.VisualStudio.Common/NuGetProjectDependencyVersionLookup.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.ProjectManagement;
+using NuGet.ProjectManagement.Projects;
+using NuGet.Versioning;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    public class NuGetProjectDependencyVersionLookup
```

Comment:
Why make it specific to build integrated projects? You can keep it generic to have all the resolved versions for each package across any kind of project. Just that in case of build integrated it will read from assets file vs for packages.config, it will be from config file.